llvm/lib/SYCLLowerIR/SYCLUtils.cpp

```diff
@@ -43,6 +47,10 @@ void traverseCallgraphUp(llvm::Function *F, CallGraphNodeAction ActionF,
         } else {
           // ... non-call is OK - add using function to the worklist
           if (auto *I = dyn_cast<Instruction>(FCall)) {
+            if (!functionFilter(I)) {
```

Comment:
`Function*` should be part of the interface, as we are deciding whether a use of particular function should be filtered or not. What if there are multiple function pointers used by `I`?